### PR TITLE
offset is clamped

### PR DESCRIPTION
**Pull request by penguin212**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/tiltedelevator/TiltedElevatorSubsystem.java [hunk 1]
```diff
@@ -269,7 +269,10 @@ public ElevatorState getState() {
      * @param power The power to change the offset by.
      */
     public void changeOffsetDistMeters(double power) {
-        this.offsetDistMeters += OFFSET_FACTOR * power;
+        // if (we are below the top OR we are going down) AND (we are above the bottom OR going up), let the driver offset
+        if((getExtensionMeters() < EXTENSION_LIMIT_METERS || power < 0) && (getExtensionMeters() > 0 || power > 0)){
```
Comment (e3l) [inline src/main/java/frc/robot/subsystems/tiltedelevator/TiltedElevatorSubsystem.java:273]: nooooooooo

File: src/main/java/frc/robot/subsystems/tiltedelevator/TiltedElevatorSubsystem.java [hunk 2]
```diff
@@ -269,7 +269,10 @@ public ElevatorState getState() {
...
+            this.offsetDistMeters += OFFSET_FACTOR * power;
```
Comment (ky28059) [inline src/main/java/frc/robot/subsystems/tiltedelevator/TiltedElevatorSubsystem.java:274]: @penguin212 what if we did a
```java
this.offsetDistMeters = MathUtil.clamp(
    offsetDistMeters = OFFSET_FACTOR,
    // distance from current extension to 0
    // distance from current extension to EXTENSION_LIMIT
);
```

**Review by ky28059 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/tiltedelevator/TiltedElevatorSubsystem.java [hunk 3]
```diff
@@ -269,7 +269,12 @@ public ElevatorState getState() {
      * @param power The power to change the offset by.
      */
     public void changeOffsetDistMeters(double power) {
-        this.offsetDistMeters += OFFSET_FACTOR * power;
+        this.offsetDistMeters = MathUtil.clamp(
+            offsetDistMeters = OFFSET_FACTOR * power + offsetDistMeters,
+            -state.getExtension(offsetState, this.pieceGrabbed),
+            EXTENSION_LIMIT_METERS - -state.getExtension(offsetState, this.pieceGrabbed)
```
Comment (e3l) [inline src/main/java/frc/robot/subsystems/tiltedelevator/TiltedElevatorSubsystem.java:275]: ? double negative ???

File: src/main/java/frc/robot/subsystems/tiltedelevator/TiltedElevatorSubsystem.java (hunk 2 above)
Comment (ky28059) [reply src/main/java/frc/robot/subsystems/tiltedelevator/TiltedElevatorSubsystem.java]: I realized I did a typo when writing out this code block in GitHub, @penguin212 can you update the `offsetDistMeters =` to a `offsetDistMeters +`? It doesn't make sense to set the offset to the offset factor 😔

**Pull request merged**